lib/cluster-def-gem/lib/cluster_def.rb

```diff
@@ -118,26 +118,35 @@ def fetch_cluster_def
         @cluster_def
       end
 
-      begin
-        # This will always fail, unless we have a node object to query
-        @cluster_def = fetch_cluster_def_http
-      rescue Exception => http_e
-        puts http_e
-        puts http_e.backtrace
-        @cluster_def = fetch_cluster_def_local
-      end
+      # This will always fail, unless we have a node object to query
+      @cluster_def = fetch_cluster_def_http
 
-      @cluster_def
+      # If @cluster_def is still nil after the http attempt, fall back.
+      @cluster_def ||= fetch_cluster_def_local
     end
 
     # fetch cluster definition via http
     def fetch_cluster_def_http
-      cluster_def_url = "http://#{@node_obj[:bcpc][:bootstrap][:server]}#{@node_obj[:bcpc][:bootstrap][:cluster_def_path]}"
-      response = Faraday.new(proxy: '').get cluster_def_url
-      if response.success? then
-        parse_cluster_def(response.body.split("\n"))
+      unless @node_obj
+        return nil
       else
-        nil
+        begin
+          cluster_def_url = 'http://' \
+            "#{@node_obj[:bcpc][:bootstrap][:server]}" \
+            "#{@node_obj[:bcpc][:bootstrap][:cluster_def_path]}"
+
+          response = Faraday.new(proxy: '').get cluster_def_url
+
+          if response.success? then
+            parse_cluster_def(response.body.split("\n"))
+          else
+            nil
+          end
+        rescue Exception => http_e
+          puts http_e
+          puts http_e.backtrace
```

Comment:
Can we print here some form of "it's fine, will fall back" or something?

Reply:
Added a warning when it falls back.